### PR TITLE
Added rust types cheat sheet

### PR DESCRIPTION
**Pull request by NateBrune**
# Goodie Cheat Sheet Pull Request Template

**What does your Instant Answer do?**
Provide a cheat sheet for rust object types

**What is the data source for your Instant Answer? (Provide a link if possible)**
https://doc.rust-lang.org/stable/reference.html#type-system

**Why did you choose this data source?**
rust-lang.org is the primary authority of the rust language specification.

**Are there any other alternative (better) data sources?**
Probably not.

**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**
Rust programmers

**Is this Instant Answer connected to a DuckDuckHack [Instant Answer idea](https://duck.co/ideas)?**
No.

**Are you having any problems? Do you need our help with anything?**
No.

**Where did you hear about DuckDuckHack? (For first time contributors)**
I am already contributor :)

**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**
img

At philly meetup. I am the 16 year old in the red shirt.

https://duck.co/ia/view/6198

### CODE REVIEW

**Comment by MariagraziaAlastra** [comment]
@NateBrune Thanks for your contribution!
Since it's not connected to an IA idea, though, we first need to evaluate it, but we'll get back to you shortly! /cc @moollaza

**Comment by moollaza** [comment]
@NateBrune it looks like the Rust Lang Cheat Sheet was claimed by an (anonymous) user. @NateBrune was that you? https://duck.co/ideas/idea/6198/rust-language-cheat-sheet

**Comment by NateBrune** [comment]
@moollaza no but the anonymous user hasnt finished the cheat sheet yet. Its a month old so I figured I was allowed to do it. Sorry if I was wrong.

**Comment by moollaza** [comment]
>Its a month old so I figured I was allowed to do it. Sorry if I was wrong.

@NateBrune I can see how this is confusing! We actually posted all these ideas a month ago so users would be able to claim them ahead of time. I've posted a message on the idea thread asking if the anon user who claimed it is interested in contributing. If no comments/PR's come before the end of the day then we'll go ahead with this PR.

Sound good?

**Comment by NateBrune** [comment]
Sounds good.

File: share/goodie/cheat_sheets/json/rust-types.json [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+                "key": "u64",
+                "val": "Unsigned integer in the range of [0, 2^64 - 1]"               
+            },
+            {
+                "key": "i8",
+                "val": "Signed integer in the range of [-(2^7), 2^7 - 1]"               
+            },
+            {
+                "key": "i16",
+                "val": "Signed integer in the range of [-(2^15, 2^15 - 1]"
```
Comment (jagtalon) [inline share/goodie/cheat_sheets/json/rust-types.json]: Missing a closing parenthesis here?

File: share/goodie/cheat_sheets/json/rust-types.json [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+                "key": "i8",
+                "val": "Signed integer in the range of [-(2^7), 2^7 - 1]"               
+            },
+            {
+                "key": "i16",
+                "val": "Signed integer in the range of [-(2^15, 2^15 - 1]"               
+            },
+            {
+                "key": "i32",
+                "val": "Signed integer in the range of [[-(2^31), 2^31 - 1]"
```
Comment (jagtalon) [inline share/goodie/cheat_sheets/json/rust-types.json]: Too many opening brackets here?

File: share/goodie/cheat_sheets/json/rust-types.json [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+                "val": "Unicode string, represented as an array of 8-bit unsigned bytes holding a sequence of UTF-8 code points."
+            }
+        ],
+        "Tuple types": [
+            {
+                "key": "tuple",
+                "val": "Tuple types and values are denoted by listing the types or values of their elements, respectively, in a parenthesized, comma-separated list."
+            }
+        ],
+        "Array, and Slice types": [
```
Comment (jagtalon) [inline share/goodie/cheat_sheets/json/rust-types.json]: Probably doesn't need a comma in here.

Comment (jagtalon) [inline share/goodie/cheat_sheets/json/rust-types.json]: Would vectors be included in this section like `Box<T>`?

**Comment by NateBrune** [comment]
There you are :+1:

**Comment by jagtalon** [comment]
@NateBrune cool! Ok will wait until the end of the day like @moollaza said. Also, what about `Vec<T>` types? Should that go with the array and slices thing?

**Comment by NateBrune** [comment]
Is that sufficient? I couldn't find an official definition.

**Comment by jagtalon** [comment]
@NateBrune looks good!

**Comment by jagtalon** [comment]
End of the day. No reply. This is good to go! :+1:

**Pull request merged**